presto-spi/src/main/java/com/facebook/presto/spi/predicate/OrExpression.java

```diff
@@ -80,6 +80,18 @@ public TupleExpression transform(Function function)
     }
 
     @Override
+    public boolean isAll()
+    {
+        return leftExpression.isAll() || rightExpression.isAll();
+    }
+
+    @Override
+    public boolean isNone()
+    {
+        return leftExpression.isNone() && rightExpression.isNone();
```

Comment:
We can assume an OrExpression to be should not have any values(`isNone()` represents that) only if both the operands have no values in their domain. So we cant use `||` here.